### PR TITLE
refactor(test): rewrite InstanceGroupManager test to insert then list

### LINKED ISSUE #129: compute.v1.it.ITComputeTest: listInstanceGroupManagersTest failed
This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: 14969009f55bc24c09c32cc9ec7ee14e8f93a081
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/cec76cee-f2d0-415f-a985-4e981dc25c8c), [Sponge](http://sponge2/cec76cee-f2d0-415f-a985-4e981dc25c8c)
status: failed

### PR DESCRIPTION
**Pull request by chingor13**
Rewrites the test to insert an `InstanceGroupManager` first, then find the one just inserted. Also splits the test out of the gigantic `ITComputeTest.java` file which we should clean up to split by client.

Implementation-wise, we create resources in the test setup and use a `MutliMap` to store self links to resources needed to clean up at the end of the test.

Fixes #129

### CODE REVIEW

File: src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  private static GlobalOperationClient globalOperationClient;
+  private static ZoneOperationClient zoneOperationClient;
+
+  private static ListMultimap<String, String> resourcesToCleanUp;
+
+  private static String instanceGroupManagerSelfLink;
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    resourcesToCleanUp = ArrayListMultimap.create();
```
Comment (elharo) [inline src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java:81]: could just initialize the field

File: src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    // list instance groups and ensure we find the one we just created
+    List<InstanceGroupManager> instanceGroupManagers =
+        Lists.newArrayList(
+            instanceGroupManagerClient.listInstanceGroupManagers(PROJECT_ZONE_NAME).iterateAll());
+    assertThat(instanceGroupManagers).isNotNull();
+    assertThat(instanceGroupManagers.size()).isGreaterThan(0);
+    boolean found = false;
+    for (InstanceGroupManager manager : instanceGroupManagers) {
+      if (manager.getSelfLink().equals(instanceGroupManagerSelfLink)) {
+        found = true;
```
Comment (elharo) [inline src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java:193]: just return here...

File: src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+      }
+    }
+    assertTrue("expected to find the inserted instance group manager", found);
```
Comment (elharo) [inline src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java:196]: and fail() if you get here
Then you don't need the variable

File: src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    boolean found = false;
+    for (InstanceGroupManagersScopedList instanceGroupManagersScopedList : managersScopedLists) {
+      List<InstanceGroupManager> managers =
+          instanceGroupManagersScopedList.getInstanceGroupManagersList();
+      if (managers == null) {
+        continue;
+      }
+      for (InstanceGroupManager manager : managers) {
+        if (manager.getSelfLink().equals(instanceGroupManagerSelfLink)) {
+          found = true;
```
Comment (elharo) [inline src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java:218]: ditto

**Review by elharo — APPROVED** [review]

File: src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java (hunk 2 above)
Comment (chingor13) [reply src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java:193]: Good point, in a previous iteration, these were in the same test so we wanted to continue. Will fix.

File: src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java (hunk 1 above)
Comment (chingor13) [reply src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java]: Done

**Pull request merged**